lib/rules/function-name-arguments-whitelist/index.js

```diff
@@ -0,0 +1,64 @@
+// @ts-nocheck
```

Comment:
We should use the [`declaration-property-unit-whitelist`](https://github.com/stylelint/stylelint/blob/master/lib/rules/declaration-property-unit-whitelist/) as a blueprint rather than [`function-url-scheme-whitelist`](https://github.com/stylelint/stylelint/blob/master/lib/rules/function-url-scheme-whitelist/) one. The latter only accepts an array of whitelisted values, whereas this rule should accept an object of function and parameter pairs (just like `declaration-property-unit-whitelist` accepts an object of property and unit pairs), .e.g.:

```json
{
  "rules": {
    "function-name-arguments-whitelist": {
      "url": ["/^images/", "^fonts/"],
      "blur": ["50%"]
    }
  }
}

```

The `declaration-property-unit-whitelist` looks for units within a declaration's value and compares it to the declaration property. You'll need to adapt the code so that this new rule looks for functions within a declaration's value, and compares the name of the function to the parameters of the function.